src/main/java/org/motechproject/prevac/service/impl/PrimeVaccinationScheduleServiceImpl.java

```diff
@@ -98,15 +102,16 @@ public PrimeVaccinationScheduleDto createOrUpdateWithDto(PrimeVaccinationSchedul
         return primeVacDtos;
     }
 
-    private Visit createScreeningVisitFromDto(PrimeVaccinationScheduleDto dto, Subject subject) {
+    private Visit createScreeningVisitFromDto(PrimeVaccinationScheduleDto dto, Subject subject, Clinic clinic) {
         Visit screeningVisit = new Visit();
         screeningVisit.setType(VisitType.SCREENING);
         screeningVisit.setSubject(subject);
         screeningVisit.setDate(dto.getActualScreeningDate());
+        screeningVisit.setClinic(clinic);
```

Comment:
find clinic by subject site id here

Reply:
Why can't we pass clinic as parameter? Of course we will get it by subject's site id.

Reply:
you can, but don't forget to check number of patients before update and create